src/base/bittorrent/sessionimpl.h

```diff
@@ -745,8 +745,8 @@ namespace BitTorrent
         QSet<QString> m_tags;
 
         // This field holds amounts of peers reported by trackers in their responses to announces
-        // (torrent.tracker_name.tracker_local_endpoint.num_peers)
-        QHash<lt::torrent_handle, QHash<std::string, QHash<TrackerEntry::Endpoint, int>>> m_updatedTrackerEntries;
+        // (torrent.tracker_name.tracker_local_endpoint.protocol_version.num_peers)
+        QHash<lt::torrent_handle, QHash<std::string, QHash<TrackerEntry::Endpoint, QHash<int, int>>>> m_updatedTrackerEntries;
```

Comment:
It probably make sense to use `std::array<int, 2>` (or `QMap<int, int>`) in place of `QHash<int, int>` but I don't insist.

Reply:
Considering that it is often used simply as an indicator (i.e. it does not contain a single element), and in most other cases it contains only one element, then I feel `std::array` is inappropriate. As for `QMap<int, int>`, does it have significant advantages? Just avoids applying a hash function to the elements?

Reply:
> As for `QMap<int, int>`, does it have significant advantages? Just avoids applying a hash function to the elements?

It's unclear from Qt docs what scenarios QMap is preferred for.

Reply:
IIRC I benchmarked QMap and QHash and found that the former is faster by a non-trivial amount when the number of element is less than ~10. That was in the Qt5 era.

Reply:
OK. Changed to `QMap`.